src/widgets/ActivityDetails/ui/ActivityDetails.tsx

```diff
@@ -20,6 +22,14 @@ export const ActivityDetailsWidget = (props: ActivityDetailsWidgetProps) => {
     activityId: "30", // Hardcoded
   })
 
+  const { setActivityDetails } = activityModel.hooks.useActivityDetailsState()
+
+  useEffect(() => {
+    if (activityDetails) {
+      setActivityDetails(activityDetails)
+    }
+  }, [activityDetails, setActivityDetails])
```

Comment:
Please use react query's onSuccess callback instead of useEffect

Reply:
Done